packages/cli-repl/src/completer.spec.ts

```diff
@@ -0,0 +1,125 @@
+import write from './completer';
+import { types as shellTypes } from 'mongosh-shell-api';
+
+import { expect } from 'chai';
+
+describe('completer.completer', () => {
+  context('when context is top level shell api', () => {
+    it('matches shell completions', () => {
+      const i = 'u';
+      expect(write(i)).to.deep.equal([['use'], i]);
+    });
+
+    it('does not have a match', () => {
+      const i = 'ad';
+      expect(write(i)).to.deep.equal([[], i]);
+    });
+
+    it('is an exact match to one of shell completions', () => {
+      const i = 'use';
+      expect(write(i)).to.deep.equal([[i], i]);
+    });
+  });
+
+  context('when context is top level db', () => {
+    // this should eventually encompass tests for DATABSE commands and
```

Comment:
There is a typo in DATABSE and inpput :). Sure you want to leave this comment here, do we have something that our process could capture?